website/integrations: owncloud: document

dominic-r opened the pull request:
sdko/feat/integration/pickup-10943-owncloud

Closes #10943 

<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://docs.goauthentik.io/docs/developer-docs/#how-can-i-contribute
-->

## Details

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->
REPLACE ME

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

Commit:
website/integrations: owncloud: init

Apply changes from old MR

Commit:
website/integrations: owncloud: easy fixes

Some easy fixes to match template

Commit:
website/integrations: owncloud: lint

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,223 @@
+> -- https://owncloud.com
+
+:::note
+This guide focuses on ownCloud installations that are deployed using Docker. If you have deployed it using a different mechanism, there may be some differences in the process.
```
```suggestion
This guide focuses on ownCloud installations that are deployed using Docker. If you deployed ownCloud using a different mechanism, there might be some differences in the process.
```

tanberry replied:
just in case you're curious about the edits, in technical writing you try to avoid using "it" and we use "might" to indicate chance... "may" is used to indicate permission. ;-)

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,223 @@
+[pre-defined](https://doc.owncloud.com/server/latest/admin_manual/configuration/user/oidc/oidc.html#client-ids-secrets-and-redirect-uris) by ownCloud for the Desktop, Android, and iOS applications.
+You are free to use the autogenerated Client ID and Secret for the Web UI (or any other appropriate ID/Secret pair).
+
+Create an application and a provider with the following settings for each of the ownCloud applications you wish to support:
```
```suggestion
Create an application and a provider for each of the ownCloud applications you wish to support.
```

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,223 @@
+
+Create an application and a provider with the following settings for each of the ownCloud applications you wish to support:
+
+### Create a provider
```
```suggestion
### Create an application and provider
```

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,223 @@
+
+### Create a provider
+
+In the authentik Admin Interface, go to **Applications -> Providers**. Create an **OAuth2/OpenID Provider** with the following parameters.
```
```suggestion
Log into authentik as an admin, navigate to **Applications --> Applications** and click **Create with Wizard**.

In the **New application** wizard, define the application details and then the provider type and settings. 

#### Application

```

tanberry replied:
Since we are using the wizard, the Application comes first, so you'll need to move all of the content from below to be up here. Then do the Provider section.

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,223 @@
+
+##### Applications
+
+#### ownCloud
```
Is this header neaded?

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,223 @@
+
+##### Service Discovery
+
+In order for the ownCloud Applications to be able to login via OIDC, your reverse proxy will have to be configured to rewrite
```
```suggestion
In order for the ownCloud Applications to be able to login via OIDC, you need to configure your reverse proxy to rewrite
```

tanberry replied:
ideally, we would say "... configure your reverse proxy to rewrite the <??> file/setting/parm from:"

In other words, where is this setting? What file?

tanberry commented on website/integrations/services/owncloud/index.md:
````diff
@@ -0,0 +1,223 @@
+  traefik.http.middlewares.owncloud-oidc-rewrite.replacepathregex.replacement: /index.php/apps/openidconnect/config
+```
+
+##### ownCloud OIDC Plugin
````
Is this a required step, or optional, or just info about this plug-in? If a required step, let's make the header title more task-specific, like "Install and configure the ownCloud OIDC Plugin" or some such.

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
website/integrations: owncloud: cleanup authentik configuration section

Cleans up the authentik configuration section of this documentation and removes unneeded bits

Commit:
website/integrations: owncloud: adjust authentik configuration headers

Add `### Configuration` header and switch General Settings, Protocl Settings, and Advanced Protocol Settings to H4

Commit:
website/integrations: owncloud: update service discovery section & remove unneeded block

Updates the service discovery block to be more in-line with documentation, renames "Apache" to "Apache HTTPD" (correct name), removes provider specific documentation for traefik and instead tells users to view provider-specific docs.

Also removes section kept last commit

Commit:
website/integrations: owncloud: start cleanup of "ownCloud configuration" section

Starts cleaning up the steps, adds warning for sub and the other one, grammar, styling and bla bla bla

Commit:
website/integrations: owncloud: fix php

Adds single quotes for client-id and client-secret lines. Should be fine I think. Logic says quotes but old docs had none

Commit:
i really don't care about a broken anchor netlify

Commit:
a

Commit:
website/integrations: owncloud: revamp "you're done section"

Cleanup and restructure the section

Commit:
website/integrations: owncloud: finish touch-ups

Cleanup the rest of the ownCloud section and update the loginButtonName variable

Commit:
website/integrations: owncloud: lint

Lints the code with Prettier

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
website/integrations: owncloud: lint

again (?)

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,217 @@
+
+The configuration for each application is nearly identical, except for the **Client ID** and **Client Secret**, which are [predefined](https://doc.owncloud.com/server/latest/admin_manual/configuration/user/oidc/oidc.html#client-ids-secrets-and-redirect-uris) by ownCloud for the Desktop, Android, and iOS applications.
+
+Follow these steps to configure authentik:
```
Wheee you do have an eye for detail. ;-) I think each of your instincts/points is valid, please do make those changes. 
Two things, though:

- On the 4 diff apps, maybe: "[…] four times, once each for the Desktop application, the Web UI, […]".  So just using "once each" at the beginning...
- Sorry for any confusion about the formatting, but let's use this: "In the Admin interface, navigate to **blah balh**. So let's not bold the "Admin interface", and use lowercase interface.

Commit:
website/integrations: owncloud: v2 authentik configuration section

Applies Tana's suggested format/layout with my suggested modifications. Lint will probably fail so enjoy this nice red cross on the right of this commit :)

Signed-off-by: 4d62 <git@sdko.org>

Commit:
website/integrations: owncloud: lint

Yea... That's about it

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Update index.md

Signed-off-by: 4d62 <git@sdko.org>

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+- `authentik.company` is the FQDN of the authentik installation.
+
+:::info
+This guide will focus on setting up ownCloud using the [official OIDC plugin](https://github.com/owncloud/openidconnect). Advanced configuration details can be found in the [ownCloud admin manual](https://doc.owncloud.com/server/latest/admin_manual/configuration/user/oidc/oidc.html).
```
```suggestion
This guide focuses on setting up ownCloud using the [official OIDC plugin](https://github.com/owncloud/openidconnect). Advanced configuration details can be found in the [ownCloud admin manual](https://doc.owncloud.com/server/latest/admin_manual/configuration/user/oidc/oidc.html).
```

Commit:
Update website/integrations/services/owncloud/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <git@sdko.org>

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+- `authentik.company` is the FQDN of the authentik installation.
+
+:::info
+This guide focuses on setting up ownCloud using the [official OIDC plugin](https://github.com/owncloud/openidconnect). Advanced configuration details can be found in the [ownCloud admin manual](https://doc.owncloud.com/server/latest/admin_manual/configuration/user/oidc/oidc.html).
```
```suggestion
This guide focuses on ownCloud installations that are deployed using Docker. If you deployed ownCloud using a different mechanism, there might be some differences in the process. Furthermore, these instructions use the [official OIDC plugin](https://github.com/owncloud/openidconnect). Advanced configuration details can be found in the [ownCloud admin manual](https://doc.owncloud.com/server/latest/admin_manual/configuration/user/oidc/oidc.html).
```

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+>
+> -- https://owncloud.com
+
+:::note
```
Let's remove that whole note... it belongs with the other note, and fits fine down in the Preparation section.

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+
+### Create applications/providers
+
+Follow these steps to create the required application/provider pairs. You will need to repeat the process four times: once for the Desktop application, once for the Web UI, once for the Android application, and once for the iOS application.
```
I prefer not repeating "once" so many times.. would suggest "You will need to repeat the process four times: once each for the Desktop application, the Web UI, the Android application, and the iOS application."

But up to you!

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+Follow these steps to create the required application/provider pairs. You will need to repeat the process four times: once for the Desktop application, once for the Web UI, once for the Android application, and once for the iOS application.
+
+1. In the Admin interface, navigate to **Applications** > **Applications**.
+2. Use the new application wizard to create the application and provider, with the following settings:
```
```suggestion
2. Use the wizard to create the application and provider, with the following settings:
```

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+    - **Configure the Provider**: Provide a name (the default name will suffice for most users), the authorization flow to use for this provider, and the following required configurations. The settings for each of the four application/provider pairs are shown below:
+
+        - **Protocol settings:**
+          **Web UI:**
```
```suggestion
          
          **Web UI:**
```

tanberry replied:
The deploy preview shows the "Web UI" smashed up against "Protocol settings". A empty line between should fix it but please verify.

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+
+    - **Configure Bindings** (optional): To manage the listing and access to applications on a user's **My applications** page, you can optionally create a [binding](https://docs.goauthentik.io/docs/add-secure-apps/flows-stages/bindings/) between the application and a specific policy, group, or user. Keep in mind that if you do not specify any bindings, all users will have access to the applicati ...
+
+### Service discovery
```
Is there a more descriptive title? Also... can we specify that it is is the reverse proxy is configured within authentik? Should that be one of the regular steps in the "authentik configuration" section?

tanberry replied:
One *can* configure a reverse proxy within authentik (@BeryJu ??), but if for sure this one is not configured there, then let's leave as is.

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+
+To enable OIDC functionality in ownCloud, follow these steps:
+
+1. **Navigate to the Market**:
```
```suggestion
1. Navigate to the **Marketplace**:
```

tanberry replied:
If I can google correctly, Market is the app used to access Marketplace... so I think we want to say Marketplace here. ?

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+
+    - Through the `config.php` file, or
+    - By storing the configuration in the ownCloud database.
+
```
That's a kinda unconventional way to show two choices, but the bullet points are good and make it easy to see the two.

I'd suggest this though really we don't need the "OR":

```
The OIDC plugin cannot be configured via the ownCloud UI. Configuration must be performed either:

    - by editing the `config.php` file
    OR
    - by storing the configuration in the ownCloud database
```

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,210 @@
+
+    The location of the `config.php` file depends on your deployment method. Consult the setup guide for your chosen deployment method to identify the file’s location within your installation.
+
+    :::note
```
Why indented?

tanberry replied:
yeah, lets not indent. it hurts my brain. :-)

Commit:
Update website/integrations/services/owncloud/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <git@sdko.org>

Commit:
Update website/integrations/services/owncloud/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <git@sdko.org>

Commit:
Update website/integrations/services/owncloud/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <git@sdko.org>

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
website/integrations: owncloud: remove duplicate

Commit:
websites/integrations: owncloud: rewrite sentence for clarity

Commit:
website/integrations: owncloud: better OR for how to config oidc

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Merge branch 'goauthentik:main' into sdko/feat/integration/pickup-10943-owncloud

Commit:
Fix indent

Signed-off-by: 4d62 <git@sdko.org>

Commit:
Lint

tanberry commented on website/integrations/services/owncloud/index.md:
```diff
@@ -0,0 +1,209 @@
+
+To support all ownCloud applications, multiple application/provider pairs are required—one each for the Web UI, Desktop application, Android application, and iOS application.
+
+The configuration for each application is nearly identical, except for the **Client ID**, **Client Secret**, and the **Redirect URI**, which are [predefined](https://doc.owncloud.com/server/latest/admin_manual/configuration/user/oidc/oidc.html#client-ids-secrets-and-redirect-uris) by ownCloud for the Desktop, Android, and iOS applications.
```
```suggestion
The configuration for each application is nearly identical, except for the **Client ID**, **Client Secret**, and the **Redirect URI** values, which are [predefined](https://doc.owncloud.com/server/latest/admin_manual/configuration/user/oidc/oidc.html#client-ids-secrets-and-redirect-uris) by ownCloud for the Desktop, Android, and iOS applications.
```

Commit:
Update website/integrations/services/owncloud/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <git@sdko.org>

tanberry reviewed (approved):
Thanks @4d62 for this PR, and thank you also @rickyelopez for the original PR!
